packages/api/src/validation/orbitAPIValidator.ts

```diff
@@ -0,0 +1,87 @@
+import { ValidatableSpec } from "@withorbit/api/src/orbitAPI";
+import OrbitAPISchema from "@withorbit/api/src/orbitAPISchema.json";
+import Ajv, {
+  ErrorObject as AjvErrorObject,
+  ValidateFunction as AjvValidationFunction,
+} from "ajv";
+import {
+  APIValidatorRequest,
+  APIValidatorError,
+  APIValidator,
+} from "./apiValidator";
+
+type OrbitAPIValidatorConfig = {
+  mutateWithDefaultValues: boolean;
+  allowUnsupportedRoute: boolean;
+};
+
+export class OrbitAPIValidator implements APIValidator {
```

Comment:
I'd love to see some tests on this when you get a chance! I have trouble being fully confident in e.g. the implementation of `allowUnsupportedRoute`, since it depends on details of ajv's validation output.

Reply:
Good call, on my list! I tried setting up jest for api package earlier but I was getting some weird jest/build errors and was planning on looping back 🙂 Thanks for the reminder!